docs/src/main/asciidoc/cdi.adoc

```diff
@@ -0,0 +1,361 @@
+A synthetic bean is not backed by a class/method/field.
+Application developers usually only consume this kind of beans.
+
+== _Q: OK injection looks cool. What other services are provided?_
+
+=== Lifecycle Callbacks
+
+A bean class may declare lifecycle `@PostConstruct` and `@PreDestroy` callbacks:
+
+.Lifecycle Callbacks Example
+[source,java]
+----
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+@ApplicationScoped
+public class Translator {
+
+    @PostConstruct <1>
+    void init() {
+       // ...
+    }
+    
+    @PreDestroy <2>
+    void destroy() {
+      // ...
+    }
+}
+----
+<1> This callback is invoked before the bean instance is put into service. It is safe to perform some initialization here.
+<2> This callback is invoked before the bean instance is destroyed. It is safe to perform some cleanup tasks here.
+
+TIP: It's a good practice to keep the logic in the callbacks "without side effects", i.e. you should avoid calling other beans inside the callbacks.
+
+=== Interceptors
+
+Interceptors are used to separate cross-cutting concerns from business logic. 
+There is a separate specification - Java Interceptors - that defines the basic programming model and semantics.
+
+.Simple Interceptor Example
+[source,java]
+----
+import javax.interceptor.Interceptor;
+import javax.annotation.Priority;
+
+@Logged <1>
+@Priority(2020) <2>
+@Interceptor <3>
+public class LoggingInterceptor {
+
+   @Inject <4>
+   Logger logger;
+
+   @AroundInvoke <5>
+   Objec logInvocation(InvocationContext context) {
+      // ...log before
+      Objec ret = context.proceed(); <6>
+      // ...log after
+      return ret;
+   }
+    
+}
+----
+<1> This is an interceptor binding annotation that is used to bind our interceptor to a bean. Simply annotate a bean class with `@Logged`.
+<2> `Priority` enables the interceptor and affects the the interceptor ordering. 
+<3> Marks an interceptor component.
+<4> An interceptor instance may be the target of dependency injection.
+<5> `AroundInvoke` denotes a method that interposes on business methods.
+<6> Proceed to the next interceptor in the interceptor chain or invoke the intercepted business method.
+
+NOTE: Instances of interceptors are dependent objects of the bean instance they intercept, i.e. a new interceptor instance is created for each intercepred bean.
+
+=== Events and Observers
+	
+Beans may also produce and consume events to interact in a completely decoupled fashion.
+Any Java object can serve as an event payload.
+The optional qualifiers act as topic selectors.
+
+.Simple Event Example
+[source,java]
+----
+
+class TaskCompleted {
+  // ...
+}
+
+@ApplicationScoped
+class ComplicatedService {
+
+   @Inject
+   Event<Task> event; <1>
+
+   void doSomething() {
+      // ...
+      event.fire(new TaskCompleted()); <2>
+   }
+
+}
+
+@ApplicationScoped
+class Logger {
+
+   void onTaskCompleted(@Observes TaskCompleted task) { <3>
```

Comment:
I would add a second sample which will use qualifiers to demonstrate the typesafe resolution for observers.

Reply:
IMHO qualifiers are not used that often together with events/observers. I'd prefer to keep it simple. I will add a link to the weld docs though.